### ISSUE TITLE
Parallel requests - random Token Signature could not be verified

### ISSUE THREAD

**Issue by majomix** [tymondesigns/jwt-auth#819]
Hello,

I'm running latest version of jwt-auth (tymon/jwt-auth dev db99bbf5c54c9792286244096c46781fae2e3b67, namshi/jose 7.2.1) and got into some strange problems. I'm using HS256 and I'm randomly getting TokenInvalidException: Token Signature could not be verified at Tymon\JWTAuth\Providers\JWT\Namshi->decode('eyJ0eXAiOiJKV1Q...').

What I mean by randomly is that I can make 300 consecutive requests to the same API out of which 1 or 2 fail. When I pick up the token from the failed request and use it again to make the same request, it works no problem. I got blacklisting disabled as it caused me some additional trouble with again seemingly radom "Token blacklisted" even for fresh tokens and even with 30 seconds grace period.

Is there any direction anyone could point me towards? I couldn't find any topic around here with a similar problem. If samples of anything would be helpful, let me know. Thanks a lot.

**Comment by majomix** [comment]
Update: I tried to reproduce the problem. I ran Postman Runner with 1000 consecutive requests to the same API with the same token, all of them ran fine. Then I tried running 3 parallel instances of Postman Runner with 1000 consecutive requests and results were consistent, 50-60 failures out of 1000. I really only have basic setup, first tried AuthenticateAndRenew then Authenticate flow but they behave the same. I'm using bundled providers and pretty much did nothing apart from generating secret and putting the middleware into my routes, What am I possibly doing wrong?
